### ISSUE TITLE
[Bug] Fenix crashes when device has a global font size other than 1.0

### ISSUE THREAD

**Issue by boek** [mozilla-mobile/fenix#2762]

**Comment by boek** [comment]
Stacktrace
```
 java.lang.IllegalStateException: Not allowed when automatic font size adjustment is enabled
        at org.mozilla.geckoview.GeckoRuntimeSettings.setFontSizeFactor(GeckoRuntimeSettings.java:826)
        at mozilla.components.browser.engine.gecko.GeckoEngine$settings$1.setFontSizeFactor(GeckoEngine.kt:192)
        at mozilla.components.browser.engine.gecko.GeckoEngine.<init>(GeckoEngine.kt:209)
        at mozilla.components.browser.engine.gecko.GeckoEngine.<init>(GeckoEngine.kt:36)
        at org.mozilla.fenix.components.Core$engine$2.invoke(Core.kt:81)
        at org.mozilla.fenix.components.Core$engine$2.invoke(Core.kt:41)
        at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
```

**Comment by buttercookie42** [comment]
This doesn't seem  quite right, because if I'm not mistaken, this would introduce inconsistent behaviour depending on whether the 100 % are encountered on startup, or through the pref change listener when the user changes the value from the settings, wouldn't it?

**Comment by buttercookie42** [comment]
Seems like a bug in android-components in that it's attempting to set the font size even though
a) auto font size setting is enabled and
b) Fenix didn't even explicitly specify a font size in it's engine settings.

**Comment by jonalmeida** [comment]
From [the comments in a-c](https://github.com/mozilla-mobile/android-components/pull/3069) when the settings was added, we probably want to check if `automaticFontSizeAdjustment` is set before changing the `fontSizeFactor` instead of always having it as false, right?

**Comment by csadilek** [comment]
I am preparing a fix for this in Android Components. The default values shouldn't cause applications to crash, unless the application configured invalid defaults (which is not the case here).

**Comment by csadilek** [comment]
Fix landed in A-C. With the latest snapshots `automaticFontSizeAdjustment` only needs to be set to `false` if either `fontSizeFactor` or `fontInflationEnabled` is also set by the application.

**Comment by ekager** [comment]
Closing as the underlying crash should be fixed from the comment above and #2795 should finish up this feature
